### PR TITLE
Regex-Check for KML

### PR DESCRIPTION
**Pull request by nadine-piveteau**
Goal: Regex test for KML
KML have been tested based on two regex. Respectively, unittests have been implemented. 

Related to [issue #3942](https://github.com/geoadmin/mf-geoadmin3/pull/3942/files)

### CODE REVIEW

File: chsdi/lib/decorators/__init__.py [hunk 1]
```diff
@@ -49,6 +50,13 @@ def wrapper(self, *args, **kwargs):
             data = urllib.unquote_plus(request.body)
             if len(data) > MAX_FILE_SIZE:
                 raise exc.HTTPRequestEntityTooLarge('File size exceed %s bytes' % MAX_FILE_SIZE)
+
+            # Prevent erroneous kml
+            print "---------------------------------------------------------"
+            print data
```
Comment (loicgasser) [inline chsdi/lib/decorators/__init__.py]: please remove the print

File: chsdi/lib/decorators/__init__.py [hunk 2]
```diff
@@ -49,6 +50,13 @@ def wrapper(self, *args, **kwargs):
...
+            data = re.sub('(\s+on\w*=(\"[^\"]+\"|\'[^\']+\'))', ' ', data, flags = re.I | re.M)
+            data = re.sub('(<|&lt;)script\s*\S*[^(>|&gt;)]*?(>|&gt;)(.|\s)*?(<|&lt;)\/script(>|&gt;)', ' ', data, flags = re.I | re.M)
+            print data
```
Comment (loicgasser) [inline chsdi/lib/decorators/__init__.py]: same here

File: chsdi/tests/integration/test_file_storage.py [hunk 3]
```diff
@@ -93,6 +152,52 @@ def test_file_too_big_kml(self):
...
+        file_id = resp.json['fileId']
+
+        # check kml based on invalid regex 1
+        resp = self.testapp.get('/files/%s' % file_id, headers=self.headers, status=200)
+        orig_data = resp.body
+        self.assertEqual(orig_data, VALID_KML)
+
+        # TEST: REGEX OCCURS SEVERAL TIMES
+        # First request, to get ids
+        resp = self.testapp.post('/files', INVALID_KML1, headers=self.headers, status=200)
```
Comment (loicgasser) [inline chsdi/tests/integration/test_file_storage.py]: Why an invalid KML repond with 200?

File: chsdi/lib/decorators/__init__.py [hunk 4]
```diff
@@ -49,6 +50,13 @@ def wrapper(self, *args, **kwargs):
             data = urllib.unquote_plus(request.body)
             if len(data) > MAX_FILE_SIZE:
                 raise exc.HTTPRequestEntityTooLarge('File size exceed %s bytes' % MAX_FILE_SIZE)
+
+            # Prevent erroneous kml
+            print "---------------------------------------------------------"
+            print data
+            data = re.sub('(\s+on\w*=(\"[^\"]+\"|\'[^\']+\'))', ' ', data, flags = re.I | re.M)
+            data = re.sub('(<|&lt;)script\s*\S*[^(>|&gt;)]*?(>|&gt;)(.|\s)*?(<|&lt;)\/script(>|&gt;)', ' ', data, flags = re.I | re.M)
```
Comment (loicgasser) [inline chsdi/lib/decorators/__init__.py]: you should test `data` to know if the kml is valid or not.

File: chsdi/tests/integration/test_file_storage.py (hunk 3 above)
Comment (gjn) [reply chsdi/tests/integration/test_file_storage.py]: We decided to silently remove the probably malicious code and keep the rest.

We are doing the same in the front-end.

Comment (loicgasser) [reply chsdi/tests/integration/test_file_storage.py]: ok got it, then just remove the prints in the services and add #NOQA as suggested and it's good to merge.

**Comment by loicgasser** [comment]
Thanks!

**Comment by gjn** [comment]
!

**Pull request merged**